compositor_common/src/validators.rs

```diff
@@ -0,0 +1,77 @@
+use std::{collections::HashSet, sync::Arc};
+
+use log::info;
+
+use crate::scene::{NodeId, SceneSpec};
+
+#[derive(Debug, thiserror::Error)]
+pub enum SpecValidationError {
+    #[error("missing node with id {0} used in transformation {1} is not defined in scene and it was not registered as an input")]
+    MissingInputNodeForTransformation(Arc<str>, Arc<str>),
+    #[error("missing node with id {0} used in output {1} is not defined in scene and it was not registered as an input")]
+    MissingInputNodeForOutput(Arc<str>, Arc<str>),
```

Comment:
I think it would be more readable if we would use enum variants with named fields